cli/src/test/java/com/box/l10n/mojito/cli/command/ThirdPartySyncCommandTest.java

```diff
@@ -1,32 +1,95 @@
 package com.box.l10n.mojito.cli.command;
 
 import com.box.l10n.mojito.cli.CLITestBase;
+import com.box.l10n.mojito.cli.utils.PollableTaskJobMatcher;
+import com.box.l10n.mojito.cli.utils.TestingJobListener;
 import com.box.l10n.mojito.entity.Repository;
-import com.box.l10n.mojito.rest.client.ThirdPartySync;
+import com.box.l10n.mojito.json.ObjectMapper;
+import com.box.l10n.mojito.rest.ThirdPartySyncAction;
+import com.box.l10n.mojito.service.thirdparty.ThirdPartySyncJob;
+import com.box.l10n.mojito.service.thirdparty.ThirdPartySyncJobInput;
+import org.junit.After;
+import org.junit.Before;
 import org.junit.Test;
-import org.slf4j.Logger;
-import org.slf4j.LoggerFactory;
+import org.quartz.JobKey;
+import org.quartz.Matcher;
+import org.quartz.Scheduler;
+import org.quartz.SchedulerException;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
 
 import java.util.Arrays;
+import java.util.List;
 
-import static org.junit.Assert.assertEquals;
-import static org.junit.Assert.assertTrue;
+import static org.assertj.core.api.Assertions.assertThat;
 
 public class ThirdPartySyncCommandTest extends CLITestBase {
 
-    /**
-     * logger
-     */
-    static Logger logger = LoggerFactory.getLogger(ThirdPartySyncCommandTest.class);
+    @Autowired
+    @Qualifier("fail_on_unknown_properties_false")
+    ObjectMapper objectMapper;
+
+    @Autowired
+    Scheduler scheduler;
+
+    Matcher<JobKey> jobMatcher;
+    TestingJobListener testingJobListener;
+
+    @Before
+    public void setUp() throws SchedulerException {
+        testingJobListener = new TestingJobListener(objectMapper);
+        jobMatcher = new PollableTaskJobMatcher<>(ThirdPartySyncJob.class);
+        scheduler.getListenerManager().addJobListener(testingJobListener, jobMatcher);
```

Comment:
We add a `JobListener` that applies solely for jobs of the `ThirdPartySyncJob` class